optimism/src/lookups.rs

```diff
@@ -71,7 +72,54 @@ pub(crate) trait FixedLookupTables<F> {
     fn table_byte() -> LookupTable<F>;
 }
 
-impl<F: Field> FixedLookupTables<F> for LookupTable<F> {
+impl<F: Field> FixedLookupTables<F, LookupTableIDs> for LookupTable<F> {
+    fn in_table(id: LookupTableIDs, value: Vec<F>) -> Option<usize> {
+        let table = match id {
+            LookupTableIDs::RangeCheck16Lookup => Self::table_range_check_16().entries,
+            LookupTableIDs::SparseLookup => Self::table_sparse().entries,
+            LookupTableIDs::ResetLookup => Self::table_reset().entries,
+            LookupTableIDs::RoundConstantsLookup => Self::table_round_constants().entries,
+            LookupTableIDs::PadLookup => Self::table_pad().entries,
+            LookupTableIDs::ByteLookup => Self::table_byte().entries,
+            _ => return None,
+        };
+        let idx = value[0]
+            .to_bytes()
+            .iter()
+            .rev()
+            .fold(0u64, |acc, &x| acc * 256 + x as u64) as usize;
+
+        match id {
+            LookupTableIDs::RangeCheck16Lookup
+            | LookupTableIDs::ResetLookup
+            | LookupTableIDs::ByteLookup
+            | LookupTableIDs::RoundConstantsLookup => {
+                if table[idx] == value {
```

Comment:
So the table must contain also the index as a value? We are using the equality on vectors here. Just for security, in case `==` is not implemented as the equality on values on each element of the vector, I would use something like:
```
table[idx].into_iter().zip(value.into_iter().for_all(|x, y| F::eq(x, y))
```

A small test is welcome :).

Reply:
I think this design can be generalized later.

Reply:
> So the table must contain also the index as a value?

Well, it doesn't have to, but it is convenient for these tables in particular. Otherwise you need to look into the whole table and that could be costlier. 

Agree that I should add some tests for the `in_table()` function, just to make sure that equality in vectors of field elements works as expected.